sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -561,7 +561,7 @@ class Dataset[T] private[sql](
    * @since 1.6.0
    */
   def schema: StructType = sparkSession.withActive {
-    queryExecution.analyzed.schema
+    removeInternalMetadata(queryExecution.analyzed.schema)
```

Comment:
@zhengruifeng for such a breaking change, we should send an email to the dev list before merging the PR.